src/logging.h

```diff
@@ -163,4 +166,89 @@ static inline void LogPrint(const BCLog::LogFlags& category, const Args&... args
     }
 }
 
+
+namespace BCLog {
+
+//! RAII-style object that outputs timing information to logs.
+template <typename TimeType>
+class Timer
+{
+public:
+    //! Log prefix; usually the name of the function this was created in.
+    std::string m_prefix{};
+
+    //! A descriptive message of what is being timed.
+    std::string m_title{};
+
+    //! Forwarded on to LogPrint if specified - has the effect of only
+    //! outputing the timing log when a particular debug= category is specified.
+    BCLog::LogFlags m_log_category{};
```

Comment:
style-nit: Any reason the members are not const and private?

Reply:
Nah, will change if rebase happens.